lib/checkbufferoverrun.cpp

```diff
@@ -358,7 +358,7 @@ static bool checkMinSizes(const std::vector<Library::ArgumentChecks::MinSize> &m
         case Library::ArgumentChecks::MinSize::ARGVALUE:
             if (Token::Match(argtok, "%num% ,|)")) {
                 const MathLib::bigint sz = MathLib::toLongNumber(argtok->str());
-                if (sz > arraySize)
+                if (sz > static_cast<MathLib::bigint>(arraySize))
```

Comment:
in general this is not a good fix. imagine that arraySize can have arbitrary values. If arraySize is very large then the cast will make it negative. If sz is 1 then your cast will cause a bug, the condition is wrong. A proper fix for this compiler warning would be to write:

    if (sz > 0 && sz > arraySize)

Imho, the cast is the wrong fix.. but the compiler will warn about this proper code so it encourages that you have conversion bugs in your code.